src/lib/tasks/Render.ts

```diff
@@ -27,17 +28,57 @@ export class RenderTask extends Task<RenderTaskParams> {
     }
     this.metrics.goto = Date.now() - start;
 
-    /* Transforming */
     const statusCode = response.status();
-    const baseHref = `${url.protocol}//${url.host}`;
-    await page!.evaluate(injectBaseHref, baseHref);
+    const headers = response.headers();
 
     start = Date.now();
     if (statusCode === 200 && minWait) {
       await page!.waitForTimeout(minWait - (Date.now() - total));
     }
     this.metrics.minWait = Date.now() - start;
 
+    if (page!.url() !== url.href) {
+      this.results = {
+        statusCode,
+        headers,
+        resolvedUrl: page!.url(),
+      };
+      return;
+    }
+
+    try {
+      const metaRefreshElement = await page!.$('meta[http-equiv="refresh"]');
+      if (metaRefreshElement) {
+        console.log(await metaRefreshElement.jsonValue());
+        const metaRefreshContent = await metaRefreshElement.getProperty(
+          'content'
+        );
+        const refreshContent = await metaRefreshContent?.jsonValue<string>();
+        const match = refreshContent?.match(/\d+;\s(?:url|URL)=(.*)/);
+        if (match) {
+          const matchedURL = match[1].replace(/'/g, ''); // Sometimes URLs are surrounded by quotes
+          const redirectURL = matchedURL.startsWith('/')
+            ? `${baseHref}${matchedURL}`
+            : match[1];
+          console.log(`Meta refresh found. Redirecting to ${redirectURL}...`);
+          this.results = {
+            statusCode,
+            headers,
+            resolvedUrl: redirectURL,
+          };
+          return;
+        }
+      }
+    } catch (e) {
+      console.log(
+        'Error while trying to check for meta[http-equive="refresh"]',
+        e
+      );
```

Comment:
should be reported instead

Reply:
Where and how?

Reply:
I thought we had sentry in this, but we don't. 
We should note that in Jira imo

Reply:
The repo is public :)

Reply:
and ? env var can be passed at runtime and we already have datadog inside

Reply:
Ok I didn't know, I liked the idea to not bloat the project too much.